File: 프로그래머스/기초 트레이닝 - Level 0/조건에 맞게 수열 변환하기2-ex.py
def solution(arr):
    answer = 0
    before = arr
    
    while True:
        after = []
        
        for num in before:
            if num >= 50 and num % 2 == 0:
                num //= 2
            elif num < 50 and num % 2 == 1:
                num = num * 2 + 1
            after.append(num)
            
        if before == after:
            break
        else:
            before = after
            answer += 1
    
    return answer

File: 프로그래머스/기초 트레이닝 - Level 0/조건에 맞게 수열 변환하기2-ex.java
import java.util.Arrays;

class Solution {
    public int solution(int[] arr) {
        return Arrays.stream(arr)
                .map(i -> count(i))
                .max()
                .getAsInt();
    }
    
    private int count(int n) {
        int cnt = 0;
        boolean flag = true;
        
        while (flag) {
            if (n % 2 == 0 && 50 <= n) {
                n /= 2;
                cnt++;
            } else if (n % 2 == 1 && n < 50) {
                n = n * 2 + 1;
                cnt++;
            } else {
                flag = false; 
            }
        }
        
        return cnt;
    }
}